components/net/bluetooth_thread.rs

```diff
@@ -395,44 +401,79 @@ impl BluetoothManager {
         sender.send(message).unwrap();
     }
 
-    pub fn get_primary_service(&mut self, device_id: String, sender: IpcSender<BluetoothObjectMsg>) {
+    pub fn get_primary_service(&mut self, device_id: String, uuid: String, sender: IpcSender<BluetoothObjectMsg>) {
         let mut adapter = match self.get_adapter() {
             Some(a) => a,
             None => send_error!(sender, "No adapter found"),
         };
+        let primary_service = match self.get_gatt_service_by_uuid(&mut adapter, &device_id, &uuid) {
+            Some(s) => s,
+            None => send_error!(sender, "No primary service found")
+        };
+        if primary_service.is_primary().unwrap_or(false) {
+            let message = BluetoothObjectMsg::BluetoothService {
+                uuid: primary_service.get_uuid().unwrap_or("".to_owned()),
+                is_primary: true,
+                instance_id: primary_service.get_object_path()
+        };
+        sender.send(message).unwrap();
+        return;
+        };
+        send_error!(sender, "No primary service found");
+    }
 
+    pub fn get_primary_services(&mut self, device_id: String,
+                                uuid: Option<String>, sender: IpcSender<BluetoothObjectMsg>) {
+        let mut services_vec: Vec<BluetoothObjectMsg> = Vec::new();
+        let mut adapter = match self.get_adapter() {
+            Some(a) => a,
+            None => send_error!(sender, "No adapter found"),
+        };
         let services = self.get_gatt_services(&mut adapter, &device_id);
         if services.is_empty() {
-            send_error!(sender, "No service found");
+                    send_error!(sender, "No service found");
         }
-
-        for service in services {
-            if service.is_primary().unwrap_or(false) {
-                let message = BluetoothObjectMsg::BluetoothService {
-                    uuid: service.get_uuid().unwrap_or("".to_owned()),
-                    is_primary: true,
-                    instance_id: service.get_object_path()
-                };
+        match uuid {
+            Some(id) => {
+                for service in services {
+                    if service.get_uuid().unwrap_or("".to_owned()) == id && service.is_primary().unwrap_or(false) {
```

Comment:
Use 'fn eq(&self, other: &String) -> bool' instead of '== ' ?
https://doc.servo.org/std/string/struct.String.html